src/tasks/lm_parsing.py

```diff
@@ -0,0 +1,178 @@
+            allf = 0
+            for x in arr:
+                allf += len(x)
+            example_counts[split] = int(math.ceil(allf / self.max_seq_len))
+        self.example_counts = example_counts
+
+    def process_split(self, split, indexers) -> Iterable[Type[Instance]]:
+        """Process a language modeling split by indexing and creating fields.
+        Args:
+            split: (list) a single list of sentences
+            indexers: (Indexer object) indexer to index input words
+        """
+        def _make_instance(sent):
+            ''' Forward targs adds <s> as a target for input </s>
+            and bwd targs adds </s> as a target for input <s>
+            to avoid issues with needing to strip extra tokens
+            in the input for each direction '''
+            d = {}
+            d["input"] = sentence_to_text_field(sent[:-1], indexers)
+            d["targs"] = sentence_to_text_field(sent[1:], self.target_indexer)
+            d["targs_b"] = sentence_to_text_field(
+                [sent[-1]] + sent[:-2], self.target_indexer)
+            return Instance(d)
+        for sent in split:
+            yield _make_instance(sent)
+
+
+@register_task('wsj', rel_path='WSJ/')
+class WSJLanguageModelling(LanguageModelingParsingTask):
+    """ Language modeling on a PTB dataset
+    See base class: LanguageModelingTask
+    """
+
+    def load_data(self, path):
+        """Load data file, tokenize text and concat sentences to create long term dependencies.
+        Args:
+            path: (str) data file path
+        """
+        seq_len = self.max_seq_len
+        tokens = []
+        with open(path) as txt_fh:
+            for row in txt_fh:
+                toks = row.strip()
+                if not toks:
+                    continue
+                toks_v = toks.split()
+                toks = toks.split() + ["<EOS>"]
+                tokens += toks
+            for i in range(0, len(tokens), seq_len):
+                yield tokens[i:i + seq_len]
+
+
+@register_task('toronto_lm', rel_path='toronto/')
+class TorontoLanguageModelling(LanguageModelingParsingTask):
+    """ Language modeling on the Toronto Books dataset
+    See base class: LanguageModelingTask
+    """
+
+    def load_data(self, path):
+        """Load data file, tokenize text and concat sentences to create long term dependencies.
+        Args:
+            path: (str) data file path
+        """
+        seq_len = self.max_seq_len
+        tokens = []
+        with open(path) as txt_fh:
+            for row in txt_fh:
+                toks = row.strip()
+                if not toks:
+                    continue
+                toks_v = toks.split()
+                toks = toks.split() + ["<EOS>"]
+                tokens += toks
+            for i in range(0, len(tokens), seq_len):
+                yield tokens[i:i + seq_len]
+
+
+@register_task('egw_lm', rel_path='egw_corpus/')
+class EnglishgigawordLanguageModelling(LanguageModelingParsingTask):
```

Comment:
Misspelling in class name. (Model*l*ing.)

Reply:
Changed.